doc-changelog/action.yml

```diff
@@ -0,0 +1,138 @@
+name: >
+  Documentation Changelog
+
+description: |
+  Adds a newsfragment to the target repository using ``towncrier``
+  containing the title and number of the pull request.
+
+
+inputs:
+
+  # Required inputs
+
+  path-to-fragment:
+    description: >
+      Path to where the towncrier fragment is stored
+    required: true
+    type: string
+
+  pr-branch:
+    description: >
+      Name of the pull request
+    required: true
+    type: string
+
+  pr-title:
+    description: >
+      Title of the pull request
+    required: true
+    type: string
+
+  pr-number:
+    description: >
+      The pull request number
+    required: true
+    type: string
+
+  token:
+    description: >
+      GitHub token for doing a git commit & push
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.pr-branch }}
```

Comment:
No need to indicate the branch. By default, this action clones the branch in the commit that triggered the workflow.